block-node/base/src/main/java/org/hiero/block/node/base/tar/TaredBlockIterator.java

```diff
@@ -98,9 +101,17 @@ public byte[] next() {
                 case NEXT_FILE:
                     if (blockIterator.hasNext()) {
                         // get the next block from the iterator
-                        final BlockAccessor currentBlock = blockIterator.next();
-                        currentBlockBytes = currentBlock.blockBytes(format).toByteArray();
-                        currentBlockName = BLOCK_NUMBER_FORMAT.format(currentBlock.blockNumber()) + extension;
+                        final BlockAccessor currentBlockAccessor = blockIterator.next();
+                        currentBlockName = BLOCK_NUMBER_FORMAT.format(currentBlockAccessor.blockNumber()) + extension;
+                        currentBlockBytes =
+                                currentBlockAccessor.blockBytes(format).toByteArray();
+                        // close the accessor as we are done with it and we need
+                        // to free resources, but only if it isn't already closed.
+                        // @todo replace Iterator with BlockAccessorBatch for this
+                        //     class so we don't need to call close here.
+                        if (!currentBlockAccessor.isClosed()) {
```

Comment:
Should we move the `isClosed()` considerations into the implementations and allow callers to just call close()?
The would also allow using of try with patterns
```
try { final BlockAccessor currentBlockAccessor = blockIterator.next()}
....
}
```

Reply:
Sounds good, wasn't sure if this was just missing the simplification.
A close and delete is a bit sneaky though.
Could that result in unintended consequences from a dev.
I guess  this is where Java docs come in and making it clear that the given Accessor deletes on close also and then tests.